torchbenchmark/__init__.py

```diff
@@ -605,11 +608,9 @@ def load_model_by_name(model):
     assert len(models) == 1, f"Found more than one models {models} with the exact name: {model}"
```

Comment:
Should we also raise `ModelNotFoundError` if `model` is not found?

Reply:
A warning message is added. This warning is triggered when the program tries to find a model in the canary list.
Test case 1: run a canary_model with missing dependencies:
```
(pytorch) PS D:\my git repo\benchmark> python run.py gat
Traceback (most recent call last):
  File "D:\my git repo\benchmark\run.py", line 304, in <module>
    Model = load_model_by_name(args.model)
  File "D:\my git repo\benchmark\torchbenchmark\__init__.py", line 607, in load_model_by_name
    raise RuntimeWarning(f"{model} is not found in the core list. Searching canary...")
RuntimeWarning: gat is not found in the core list. Searching canary...
Traceback (most recent call last):
  File "D:\my git repo\benchmark\run.py", line 312, in <module>
    Model = load_canary_model_by_name(args.model)
  File "D:\my git repo\benchmark\torchbenchmark\__init__.py", line 626, in load_canary_model_by_name
    module = importlib.import_module(f'.canary_models.{model}', package=__name__)
  File "C:\Users\temp\anaconda3\envs\pytorch\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\my git repo\benchmark\torchbenchmark\canary_models\gat\__init__.py", line 1, in <module>
    from torchbenchmark.util.framework.gnn.model_factory import GNNModel
  File "D:\my git repo\benchmark\torchbenchmark\util\framework\gnn\model_factory.py", line 6, in <module>
    from torch_geometric.nn import GAT, GCN, GraphSAGE
ModuleNotFoundError: No module named 'torch_geometric'
```
Test case 2: run a non-exist model:
```
(pytorch) PS D:\my git repo\benchmark> python run.py abcde
Traceback (most recent call last):
  File "D:\my git repo\benchmark\run.py", line 304, in <module>
    Model = load_model_by_name(args.model)
  File "D:\my git repo\benchmark\torchbenchmark\__init__.py", line 607, in load_model_by_name
    raise RuntimeWarning(f"{model} is not found in the core list. Searching canary...")
RuntimeWarning: abcde is not found in the core list. Searching canary...
Traceback (most recent call last):
  File "D:\my git repo\benchmark\run.py", line 312, in <module>
    Model = load_canary_model_by_name(args.model)
  File "D:\my git repo\benchmark\torchbenchmark\__init__.py", line 625, in load_canary_model_by_name
    raise ModelNotFoundError(f"{model} is not found in the both core and canary model lists")
torchbenchmark.ModelNotFoundError: abcde is not found in the both core and canary model lists
```